server/src/main/java/org/elasticsearch/index/codec/bloomfilter/ES87BloomFilterPostingsFormat.java

```diff
@@ -180,33 +179,38 @@ public Iterator<String> iterator() {
         }
 
         private void writeBloomFilters(Fields fields) throws IOException {
-            for (String field : fields) {
-                final Terms terms = fields.terms(field);
-                if (terms == null) {
-                    continue;
-                }
-                final int bloomFilterSize = bloomFilterSize(state.segmentInfo.maxDoc());
-                final int numBytes = numBytesForBloomFilter(bloomFilterSize);
-                try (ByteArray buffer = bigArrays.newByteArray(numBytes)) {
+            final int bloomFilterSize = bloomFilterSize(state.segmentInfo.maxDoc());
+            final int numBytes = numBytesForBloomFilter(bloomFilterSize);
+            final int[] hashes = new int[NUM_HASH_FUNCTIONS];
```

Comment:
Local variable access is often faster than field access, no point in having a field anyway since we only call method a single time in its lifetime I think? (even if we repeatedly call it, the allocation is trivial)